Doc/library/getopt.rst

```diff
@@ -144,26 +145,65 @@ In a script, typical usage is something like this::
            else:
                assert False, "unhandled option"
-       # ...
+       process(args, output=output, verbose=verbose)
 
    if __name__ == "__main__":
        main()
 
 Note that an equivalent command line interface could be produced with less code
-and more informative help and error messages by using the :mod:`argparse` module::
+and more informative help and error messages by using the :mod:`optparse` module::
 
-   import argparse
+   import optparse
 
    if __name__ == '__main__':
-       parser = argparse.ArgumentParser()
-       parser.add_argument('-o', '--output')
-       parser.add_argument('-v', dest='verbose', action='store_true')
-       args = parser.parse_args()
-       # ... do something with args.output ...
-       # ... do something with args.verbose ..
+         parser = optparse.OptionParser()
+         parser.add_option('-o', '--output')
+         parser.add_option('-v', dest='verbose', action='store_true')
+         opts, args = parser.parse_args()
+         process(args, output=opts.output, verbose=opts.verbose)
+
+A roughly equivalent command line interface for this case can also be
+produced by using the :mod:`argparse` module::
+
+import argparse
+
+if __name__ == '__main__':
+      parser = argparse.ArgumentParser()
+      parser.add_argument('-o', '--output')
+      parser.add_argument('-v', dest='verbose', action='store_true')
+      parser.add_argument('rest', nargs='*')
+      args = parser.parse_args()
+      process(args.rest, output=args.output, verbose=args.verbose)
+
+However, unlike the ``optparse`` example, this ``argparse`` example will
+handle some parameter combinations differently from the way the ``getopt``
+version would handle them. For example (amongst other differences):
+
+* supplying ``-o -v`` gives ``output="-v"`` and ``verbose=False``
+  for both ``getopt`` and ``optparse``,
+  but a usage error with ``argparse``
+  (complaining that no value has been supplied for ``-o/--output``,
+  since ``-v`` is interpreted as meaning the verbosity flag)
+* similarly, supplying ``-o --`` gives ``output="--"`` and ``args=()``
+  for both ``getopt`` and ``optparse``,
+  but a usage error with ``argparse``
+  (also complaining that no value has been supplied for ``-o/--output``,
+  since ``--`` is interpreted as terminating the option processing
+  and treating all remaining values as positional arguments)
+* supplying ``-o=foo`` gives ``output="=foo"``
+  for both ``getopt`` and ``optparse``,
+  but gives ``output="foo"`` with ``argparse``
+  (since ``=`` is special cased as an alternative separator for
+  option parameter values)
+
+Whether these differing behaviors in the ``argparse`` version are
+considered desirable or a problem will depend on the specific command line
+application use case.
 
 .. seealso::
 
-   Module :mod:`argparse`
-      Alternative command line option and argument parsing library.
+   Module :mod:`optparse`
+      More object-oriented command line option parsing.
```

Comment:
IMO the salient feature of optparse over getopt is that it’s more convenient and automated! «more OO» is not a selling point

Reply:
I've replaced "object-oriented" with "declarative" in the places where I had used it.